### ISSUE TITLE
Enhance andFilterCompare to include ranges

### ISSUE THREAD

**Issue by Iliolou** [yiisoft/yii2#19782]
We can enhance andFilterCompare operator to include ranges .e.g 2-4 , apart from >3, <>3

```diff
         if (preg_match('/^(<>|>=|>|<=|<|=)/', $value, $matches)) {
             $operator = $matches[1];
             $value = substr($value, strlen($operator));
+        } elseif (count($matches=preg_split("/-/",$value)) == 2) {
+		return $this->andFilterWhere(['between', $name, $matches[0], $matches[1]]);
         } else {
             $operator = $defaultOperator;
         }
```

**Comment by samdark** [comment]
Interesting.

1. Adjust code.
2. Add docs.
3. Add CHANGELOG line.

Do you have some time for a pull request?

**Comment by Iliolou** [comment]
Not until next week, sorry. But if you think it's useful, just test it and add it as your own.

**Comment by samdark** [comment]
@yiisoft/reviewers what do you think?

**Comment by lubosdz** [comment]
* should do much better type checks to prevent crash, e.g. " 5 - a ", " < 56-01", " 1 -", " - 987" , " 1.123 - 1.456 " ..
* floats supported too?
* seems like nice enhancement, but I don't remember needing it
* potential BC break e.g. " <> 123-456 " ?

**Comment by Iliolou** [comment]
1) There are no crashes, it's just a filter. If it's invalid finds no records. Cases  " < 56-01", " 1 -", " - 987"  are not hanlded by my enhancement. First one starts with < and the rest do not have two arguments. Case " 5 - a " is valid and returns no records.
2) Yes, case " 1.123 - 1.456 " works as expected.
3) My customers needed it, and are actually very happy about it. That's why I though I should submit it.
4) Not handled by the enhancement, returns the same as before (only the <> 123 part is validated).

**Comment by bizley** [comment]
Yes, I think we can add it, potential mismatches will be handled by the developer anyway since this is a filter. I'll ask for the same that Samdark did + tests.

**Comment by lubosdz** [comment]
At least numeric arguments should be ensured. Remember, there are multiple RDBMS supported and simply splitting by `-` and not checking numeric type may cause BC crashes for other users.

**Comment by terabytesoftw** [comment]
Thanks for the suggestion and discussion.

After reviewing the implementation in #20905, we decided not to add automatic `min-max` parsing to `andFilterCompare()`. It would change the meaning of existing valid values and override custom default operators, introducing ambiguous backward-incompatible behavior.

Applications can parse this input explicitly and use `andFilterWhere(['between', $column, $min, $max])`. Since we do not plan to include this behavior in either `master` or `22.0`, we are closing this issue as not planned.
